app/status.go

```diff
@@ -137,7 +137,9 @@ func GetUserStatusesByIds(userIds []string) ([]*model.Status, *model.AppError) {
 		}
 	}
 	for _, userId := range missingUserIds {
-		statusMap = append(statusMap, &model.Status{UserId: userId, Status: "offline"})
+		if result := <-Srv.Store.User().Get(userId); result.Err == nil {
+			statusMap = append(statusMap, &model.Status{UserId: userId, Status: "offline"})
```

Comment:
the issue here is if you send a non-existing id then it will appear as offline.
Or we make all remaining missing id as offline or we just ignore those.
I also thought to mark the non-existing in the return message as does not exist in the status (but this maybe is make no sense)